src/components/BetterSwipeable.tsx

```diff
@@ -0,0 +1,440 @@
+   */
+  containerStyle?: StyleProp<ViewStyle>;
+
+  /**
+   * Style object for the children container (`Animated.View`), for example to
+   * apply `flex: 1`
+   */
+  childrenContainerStyle?: StyleProp<ViewStyle>;
+
+  // typescript is determined that component created with forwardRef cannot have children
+  children?: React.ReactNode;
+}
+
+export interface SwipeableController {
+  close: () => void;
+  openLeft: () => void;
+  openRight: () => void;
+  panGesture: PanGesture;
+}
+
+export const Swipeable = React.forwardRef<SwipeableController, SwipeableProps>(
+  (props, ref) => {
+    const { children, renderLeftActions, renderRightActions } = props;
+
+    const [rowState, setRowState] = useState(0); // current state: 1 - left, 0 - closed, -1 - right
+    const [rowWidth, setRowWidth] = useState(0);
+    const [leftWidth, setLeftWidth] = useState(0);
+    // when renderRightActions is not set the rightOffset is equal to 0, making rightWidth equal to
+    // rowWidth. With undefined as a starting point it is possible to determine when no right actions
+    // are set
+    const [rightOffset, setRightOffset] = useState<number | undefined>(
+      undefined
+    );
+    const rightWidth = rowWidth - (rightOffset ?? rowWidth);
+
+    const dragOffset = useSharedValue(0); // offset for the top component
+    const dragStartOffset = useSharedValue(0); // origin for pan gesture handler translation
+    const leftActionOffset = useSharedValue(0); // offset used to hide left panel when right is opened
+    const rightActionOffset = useSharedValue(0); // offset used to hide right panel when left is opened
+
+    // parameters passed to actions' render methods
+    const leftActionProgress = useDerivedValue(() => {
+      const progress = dragOffset.value / leftWidth;
+      return isNaN(progress) ? 0 : progress;
+    }, [dragOffset, leftWidth]);
+
+    const leftActionDrag = useDerivedValue(() => {
+      return dragOffset.value;
+    }, [dragOffset]);
+
+    const rightActionProgress = useDerivedValue(() => {
+      const progress = -dragOffset.value / rightWidth;
+      return isNaN(progress) ? 0 : progress;
+    }, [dragOffset, rightWidth]);
+
+    const rightActionDrag = useDerivedValue(() => {
+      return dragOffset.value;
+    }, [dragOffset]);
+
+    // default config
+    const friction = props.friction ?? 1;
+    const leftThreshold = props.leftThreshold ?? leftWidth / 2;
+    const rightThreshold = props.rightThreshold ?? rightWidth / 2;
+    const overshootLeft = props.overshootLeft ?? leftWidth > 0;
+    const overshootRight = props.overshootRight ?? rightWidth > 0;
+    const overshootFriction = props.overshootFriction ?? 1;
+    const animationOptions = props.animationOptions ?? {
+      damping: 30,
+      stiffness: 250,
+    };
+
+    // animated styles for panels
+    const childContainerStyle = useAnimatedStyle(() => {
+      return {
+        transform: [{ translateX: dragOffset.value }],
+      };
+    });
```

Comment:
```suggestion
    const childContainerStyle = useAnimatedStyle(() => ({
      transform: [{ translateX: dragOffset.value }],
    }));
```

Reply:
Done in 58ddd80a0373a6603a27376f4edc33b90eb8886d